Packs/ZoomMail/Integrations/ZoomMail/ZoomMail.py

```diff
@@ -0,0 +1,1219 @@
+    if len(err) > 0:
+        return f"Errors were found while testing:\n{''.join(err)}"
+    return "ok"
+
+
+def fetch_incidents(client: ZoomMailClient, params: dict[str, str]) -> None:
+    """
+    Fetches email messages from ZoomMail API and creates incidents based on the specified criteria.
+
+    :param client: The ZoomMailClient instance.
+    :param params: Configuration parameters including mailbox, query, fetch labels, and control flags.
+    """
+    fetch_from: Optional[str] = params.get("default_mailbox")
+    fetch_query: str = params.get("fetch_query", "")
+    first_fetch_time: str = params.get("first_fetch", "3 days")
+    fetch_labels: str = params.get("fetch_labels", "INBOX")
+    fetch_threads: Optional[bool] = params.get("fetch_threads", False)
+
+    max_fetch = min(int(params.get("max_fetch", 50)), 200)
+
+    last_run = demisto.getLastRun()
+    last_fetch_info = last_run.get("last_fetch_info", {"internalDate": 0, "ids": []})
+    last_page_fetch_token = last_run.get("next_page_token")
+
+    if not last_fetch_info["internalDate"]:
+        first_fetch_dt = parse(first_fetch_time)
+        if not first_fetch_dt:
+            demisto.info("No last fetch was found, defaulting to 3 days.")
+            first_fetch_dt = datetime.now() - timedelta(days=3)
+        last_fetch_info["internalDate"] = first_fetch_dt.timestamp()
+
+    incidents: list[dict[str, Any]] = []
+
+    query = f"{fetch_query} after:{int(last_fetch_info['internalDate'])}"
+    if last_page_fetch_token:
+        demisto.info(f"Fetching additional messages from previous fetch run, {last_page_fetch_token}")
+        messages_response = client.list_emails(
+            email=fetch_from, page_token=last_page_fetch_token, label_ids=fetch_labels
+        )
+    else:
+        messages_response = client.list_emails(
+            email=fetch_from,
+            max_results=str(max_fetch),
+            query=query,
+            label_ids=fetch_labels,
+        )
+    next_page_token = messages_response.get("nextPageToken", None)
+    messages = messages_response.get("messages", [])
+    message_dates: list[float] = []
+
+    for msg in messages:
+        message_id = msg.get("id")
+        thread_id = msg.get("threadId", "")
+        message_details = client.get_email_message(
+            email=fetch_from, message_id=message_id
+        )
+        internal_date = float(message_details.get("internalDate")) / 1000.0
+
+        msg.update({"internalDate": internal_date})
+
+        # Check if it's a new message, and if it's either all messages (fetch_threads is False) or only thread starters
+        if (internal_date > last_fetch_info["internalDate"] or (
+            internal_date == last_fetch_info["internalDate"] and message_id not in last_fetch_info["ids"]
+        )) and (not fetch_threads or message_id == thread_id):
+            incident = zoom_mail_to_incident(message_details, client, fetch_from)
```

Comment:
the following case needs unit tests:
Deduplecation of ids that exist in the last_fetch_info["ids"]
filtering of threads when fetch_threads is true, not filtering threads when it is false.